docs/DesignDocs/LoggingFormat.md

```diff
@@ -0,0 +1,79 @@
+Logging Format
+=====
+
+When using the OE SDK and in trying to make use of the logs (e.g. via uploading them to Azure Log Analytics) it is nice to be able to parse and display all of the available logs including the ones generated by OE.
+
+Motivation
+----------
+
+Users should be able to specify and control their own logging format so that they can further parse the log files as they see fit.
+
+User Experience
+---------------
+
+- The user can set the `OE_LOG_DEVICE` environmental variable to specify the log file location
+
+- The user can set the `OE_LOG_FORMAT` environmental variable to specify the log format. The user will be given the below information:
+
+        - time in string format
+        - microseconds
+        - is enclave (E|H)
+        - log level
+        - thread id
+        - log message
+        - file log initiated from
+        - function log initiated from
+        - file line number log initiated from
+
+        so for example a log foramt could be set like this:
+
+        `OE_LOG_FORMAT=$'{\"e_ts\":\"%s.%06ldZ\",\"level\":\"(%s)%s\",\"tid\":\"tid(0x%lx)\",\"msg\":\"%s\",\"file\":\"%s\",\"func\":\"%s\",\"number":\"%s\"}\n'`
+
+        and end up looking like this:
+
+        `
+        {
+                "e_ts": "2019-09-24T10:09:27.832475Z",
+                "file": "switchless/host/host.c",
+                "func": "main",
+                "level": "(H)ERROR",
+                "msg": "Hello 'world'!",
+                "number": "79",
+                "tid": "tid(0x7fc2698a6d40)"
+        }
+        `
+
+- The user can set the `OE_JSON_ESCAPE` environmental variable and if it is set then the log message will be escaped in order to be compatible with the JSON standard.
+
+Specification
+-------------
+
+Users can specify their own format when logging to a user specified file via the existing `OE_LOG_DEVICE` environmental variable.
+If the var is not specified then logging goes on as usual.
+
+If the var is specified then we look for an extra environmental variable `OE_LOG_FORMAT`.
+If `OE_LOG_FORMAT` is specified we call a method where we pass in the log line information and it is formatted by the `OE_LOG_FORMAT` string. If it is set it needs to end with a newlone.
+
+If it is not then we use the default format string to log in the file.
+
+In addition to the `OE_LOG_FORMAT` environmental variable, if the `OE_JSON_ESCAPE` environmental variable is set, then the log message will be processed accordingly in order to be compatible with the JSON standard.
+
+When logging to a separate file the normal logs are directed to stdout AND the log with the user specified format is directed to the file. This is useful if you want to keep the human readable logs, but also redirect a machine-readable format (e.g. JSON) to a file.
```

Comment:
There should be an option to have output go to file only.

Reply:
Changed it to default to file only when file is specified, but optionally we can log to both stdout and file.

Reply:
Thanks for addressing my comment